File: platforms/documentation/docs/src/snippets/reference/plugin-development/validatePluginsConfig/common/src/main/java/org/example/HelloTask.java
package org.example;

import org.gradle.api.DefaultTask;
import org.gradle.api.provider.Property;
import org.gradle.api.tasks.Input;
import org.gradle.api.tasks.TaskAction;
import org.gradle.api.tasks.UntrackedTask;

@UntrackedTask(because = "Prints a message and produces no cacheable output")
public abstract class HelloTask extends DefaultTask {

    @Input
    public abstract Property<String> getMessage();

    @TaskAction
    public void run() {
        System.out.println(getMessage().get());
    }
}

File: samples/multi-project-with-buildSrc/buildSrc/src/main/kotlin/HelloTask.kt
import org.gradle.api.*
import org.gradle.api.tasks.*
import org.gradle.kotlin.dsl.*

open class HelloTask : DefaultTask() {

    init {
        group = "My"
        description = "Prints a description of ${project.name}."
    }

    @TaskAction
    fun run() {
        println("I'm ${project.name}")
    }
}

fun Project.declareHelloTask() =
    task<HelloTask>("hello")

val Project.hello: HelloTask
    get() = tasks["hello"] as HelloTask
